### ISSUE TITLE
flush problem with the modem serial

### ISSUE THREAD

**Issue by dmnc-net** [rocketscream/Low-Power#75]
Hi, I need some help with finding the issue. I want to put my board to sleep for a bigger amount of time so I've created a function which repeats 8sec sleep command. In the example code below everything is working but in my complex project the problem occurs. First 9 cycles doesn't sleep and after that 10th call is working. Any clue what can affect this? Thanks.

```
#include <LowPower.h>

void lowPower(uint16_t len){
    for (uint8_t i = 1; i<=round(len/8+0.5); i++) {
      Serial.print(i);
      Serial.flush();
      LowPower.idle(SLEEP_8S, ADC_OFF, TIMER2_OFF, TIMER1_OFF, TIMER0_OFF, SPI_OFF, USART0_OFF, TWI_OFF);
      Serial.print("/");
      Serial.flush();
    }
}

void setup() {
  Serial.begin(38400);
  lowPower(30);
}

void loop() {
}

```

**Comment by dmnc-net** [comment]
okay
1) number 9 is not solid
2) this issue is caused by this function which does power on/off for the GSM/GPS board with SIM808:
```
#define MODEM_PWR_PIN               9
#define MODEM_PWR_IMPULSE           1500

void ATHelper::powerImpulse() {
  pinMode(MODEM_PWR_PIN, OUTPUT);
  digitalWrite(MODEM_PWR_PIN, HIGH);
  delay (MODEM_PWR_IMPULSE);
  digitalWrite(MODEM_PWR_PIN, LOW);
}

```

**Comment by fsantanna** [comment]
, to

On Oct 27, 2018 16:31, "DMNC" <notifications@github.com> wrote:

Closed #75 <https://github.com/rocketscream/Low-Power/issues/75>.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/rocketscream/Low-Power/issues/75#event-1930440809>, or mute
the thread
